lib/repository.py

```diff
@@ -29,123 +26,79 @@ def download_from_repository(filepath, newpath, repo=REPOSITORY):
         pass
 
 
-def more_recent(latest, current):
-    """Given two version number strings, returns True if the first is more recent."""
-    latest_parts = latest.split('.')
-    current_parts = current.split('.')
-    for n in range(len(latest_parts)):
-        l = latest_parts[n]
-        if len(current_parts) < (n + 1):
-            return (l != "rc")
-        c = current_parts[n]
-        if l > c:
-            return True
-        elif c > l:
-            return False
-    return (len(current_parts) > (n + 1) and current_parts[n + 1] == "rc")
-
-
 def check_for_updates():
-    """Check for updates to scripts and executable."""
-    init = InitThread()
-    init.run()
+    """Check for updates to scripts.
 
+    This updates the HOME_DIR scripts directory with the latest script versions
+    """
+    try:
+        # open version.txt for current release branch and get script versions
+        version_file = urllib.request.urlopen(REPOSITORY + "version.txt")
+        version_file.readline()
+
+        # read scripts from the repository and the checksums from the version.txt
+        scripts = []
+        print("Downloading scripts...")
+        for line in version_file:
+            scripts.append(line.decode().strip('\n').split(','))
+
+        total_script_count = len(scripts)
+
+        # create script directory if not available
+        if not os.path.isdir(SCRIPT_WRITE_PATH):
+            os.makedirs(SCRIPT_WRITE_PATH)
+
+        update_progressbar(0.0 / float(total_script_count))
+        for index, script in enumerate(scripts):
+            script_name = script[0]
+            if len(script) > 1:
+                script_version = script[1]
+            else:
+                script_version = None
+
+            path_script_name = os.path.normpath(os.path.join(HOME_DIR, "scripts", script_name))
+            if not file_exists(path_script_name):
+                download_from_repository("scripts/" + script_name,
+                                         os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)))
+
+            need_to_download = False
+            try:
+                file_object, pathname, desc = imp.find_module(''.join(script_name.split('.')[:-1]), [SCRIPT_WRITE_PATH])
+                new_module = imp.load_module(script_name, file_object, pathname, desc)
+                m = str(new_module.SCRIPT.version)
+                need_to_download = str(script_version) > m
```

Comment:
Should this comparison be done on the floating point values instead of the string versions to avoid issues like:

```
In [3]: "2.0" > "11.2"
Out[3]: True
```